### PR TITLE
Update contributing instructions

### PR DESCRIPTION
**Pull request by gaearon**
Now that we mandate Yarn.

### CODE REVIEW

File: content/docs/how-to-contribute.md [hunk 1]
```diff
@@ -95,58 +95,51 @@ In order to accept your pull request, we need you to submit a CLA. You only need
...
-* `npm run flow` runs the [Flow](https://flowtype.org/) typechecks.
-* `npm run build` creates a `build` folder with all the packages.
+* `yarn lint` checks the code style.
+* `yarn linc` is like `yarn lint` but faster because it only checks files that differ in your branch.
+* `yarn test` runs the complete test suite.
+* `yarn test --watch` runs an interactive test watcher.
+* `yarn test <pattern>` runs tests with matching filenames.
+* `yarn flow` runs the [Flow](https://flowtype.org/) typechecks.
+* `yarn build` creates a `build` folder with all the packages.
+* `yarn build core,dom --type=UMD` creates UMD builds of just React and ReactDOM.
```
Comment (bvaughn) [inline content/docs/how-to-contribute.md]: Cool ^ nicer than than the previous syntax

I guess we can remove the extra "--" from [the release script](https://github.com/facebook/react/blob/master/scripts/release/build-commands/build-artifacts.js#L9) now too?

File: content/docs/how-to-contribute.md [hunk 2]
```diff
@@ -95,58 +95,51 @@ In order to accept your pull request, we need you to submit a CLA. You only need
...
-* You are familiar with `npm` and know whether or not you need to use `sudo` when installing packages globally.
-* You are familiar with `git`.
+* You have [Node](https://nodejs.org) installed at v6.0.0+ and [Yarn](https://yarnpkg.com/en/) at v1.0.0+.
+* You have `gcc` installed or are comfortable installing a compiler if needed. Some of our dependencies may require a compilation step. On OS X, the Xcode Command Line Tools will cover this. On Ubuntu, `apt-get install build-essential` will install the required packages. Similar commands should work on other Linux distros. Windows will require some additional steps, see the [`node-gyp` installation instructions](https://github.com/nodejs/node-gyp#installation) for details.
```
Comment (gaearon) [inline content/docs/how-to-contribute.md]: This looks really scary btw. Is it really true?

File: content/docs/how-to-contribute.md (hunk 1 above)
Comment (gaearon) [reply content/docs/how-to-contribute.md]: Yea. Yarn 1.0 warns about it and says it's unnecessary.

File: content/docs/how-to-contribute.md (hunk 2 above)
Comment (bvaughn) [reply content/docs/how-to-contribute.md]: I was actually just scratching my head and wondering the same thing

Comment (gaearon) [reply content/docs/how-to-contribute.md]: Let's just delete this maybe

Comment (bvaughn) [reply content/docs/how-to-contribute.md]: Maybe it dates back to stuff like #4916?

Comment (bvaughn) [reply content/docs/how-to-contribute.md]: But yeah, I'm cool with removing it.

**Pull request merged**